docs/proposals/20210203-unmanaged-infrastructure.md

```diff
@@ -0,0 +1,236 @@
+Importantly, it will not modify the resource or its status in any way.
+A predicate will be provided in the Cluster API repository to aid provider implmentations in filtering resources that are externally managed.
+
+Additionally, the external management system must provide all required fields within the spec of the InfraCluster and must adhere to the CAPI provider contract and set the InfraCluster status to be ready when it is appropriate to do so.
+
+While an "externally managed" InfraCluster won't reconcile or manage the lifecycle of the cluster infrastructure, CAPI will still be able to create compute nodes within it.
+
+The machine controller must be able to operate without hard dependencies regardless of the cluster infrastructure being managed or externally managed.
+![](https://i.imgur.com/nA61XJt.png)
+
+### User Stories
+
+#### Story 1 - Alternate control plane provisioning with user managed infrastructure
+As a cluster provider I want to use CAPI in my service offering to orchestrate Kubernetes bootstraping while letting workload cluster operators own their infrastructure lifecycle.
+
+#### Story 2 - Restricted access to cloud provider APIs
+As a cluster operator I want to use CAPI to orchestrate kubernetes bootstraping while restricting the privileges I need to grant for my cloud provider because of organisational cloud security constraints.
+
+#### Story 3 - Consuming existing cloud infrastructure
+As a cluster operator I want to use CAPI to orchestate Kubernetes bootstraping while reusing infrastructure that has already been created in the organisation either by me or another team.
+
+### Implementation Details/Notes/Constraints
+
+**Managed**
+
+- It will be default and will preserve existing behaviour. An InfraCluster CR without the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation.
+
+
+**Externally Managed**
+
+An InfraCluster CR with the `cluster.x-k8s.io/managed-by: "<name-of-system>"` annotation.
+
+The provider InfraCluster controller must:
+- Skip any reconciliation of the resource.
+
+- Not update the resource or its status in any way
+
+The external management system must:
+
+- Populate all required fields within the InfraCluster spec to allow other CAPI components to continue as normal.
+
+- Adhere to all Cluster API contracts for infrastructure providers.
+
+- When the infrastructure is ready, set the appropriate status as is done by the provider controller today.
+
+#### Provider implementation changes
+
+To enable providers to implement the changes required by this contract, Cluster API is going to provide a new `predicates.ResourceExternallyManaged` predicate as part of its utils.
+
+This predicate filters out any resource that has been marked as "externally managed" and prevents the controller from reconciling the resource.
+
+### Security Model
+
+When externally managed, the required cloud provider privileges are significantly reduced when compared with a traditionally managed cluster.
```

Comment:
I'm not sure I understand what you mean here. By "cloud provider", do you mean k8 cloud controller manager or the infrastructure / cloud? How are the privileges "significantly reduced"?

Reply:
I think this is referring to the InfraCloud controller.
My interpretation of the reduced privileges is because the InfraCluster controller will no longer have to provision infrastructure resources and therefore will not need the corresponding permissions.